canonical-web-and-design/maas-ui
# feat(ui): build form for adding special filesystems

## Done

- Built AddSpecialFilesystem form

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine storage tab of a machine with editable storage
- Check there is an "Add special filesystem" button
- Click it, fill in the form and submit
- Check that the form closes and the special filesystem is added to the filesystems list
- Note that the form does not handle errors yet, for example when trying to mount where there is already a filesystem. This will be handled once https://github.com/canonical-web-and-design/maas-ui/issues/1842 has landed.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2161

## Screenshot
![Screenshot_2020-11-23 fun-mammal maas storage bolla MAAS](https://user-images.githubusercontent.com/25733845/99923686-9264a580-2d82-11eb-9ac3-bb4e5351dcc0.png)
